### PR TITLE
*: add succ filed to slow log and fix shallow copy problem when parse slow log file.

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
* Add `Succ` filed name to slow log and `SLOW_QUERY` table.
* change `SLOW_QUERY`,`Query` column type from `varchar(4096)` to `longtext`.
* fix shallow copy problem caused by the long query(len(query) > 4094).

### What is changed and how it works?
* Add `copy` for `getOneLine` function.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by crazycs520** [comment]
@zz-jason @zimulala PTAL

**Comment by Deardrops** [comment]
How many bytes in query string will cause the shallow copy problem?

**Comment by crazycs520** [comment]
@Deardrops When the length of query more than 4094.

File: infoschema/slow_log.go [hunk 1]
```diff
@@ -319,6 +329,12 @@ func (st *slowQueryTuple) setFieldValue(tz *time.Location, field, value string)
 			return errors.AddStack(err)
 		}
 		st.memMax = num
+	case variable.SlowLogSucc:
+		succ, err := strconv.ParseBool(value)
+		if err != nil {
+			return errors.AddStack(err)
```
Comment (zz-jason) [inline infoschema/slow_log.go]: you can use `errors.Wrap()` to explain which operation causes this error.

Comment (crazycs520) [reply infoschema/slow_log.go]: Great, thanks, Done. 
I add `errors.Wrap()` outside of this function.

Comment (zz-jason) [reply infoschema/slow_log.go]: I think it's better to wrap it here? Why delaying it to the outside?

File: infoschema/slow_log.go [hunk 2]
```diff
@@ -102,7 +102,7 @@ func ParseSlowLog(tz *time.Location, reader *bufio.Reader) ([][]types.Datum, err
 			st = &slowQueryTuple{}
 			err = st.setFieldValue(tz, variable.SlowLogTimeStr, line[len(variable.SlowLogStartPrefixStr):])
 			if err != nil {
-				return rows, err
+				return rows, errors.Wrap(err, "parse slow log filed `"+variable.SlowLogTimeStr+"` error")
```
Comment (zz-jason) [inline infoschema/slow_log.go]:
```suggestion
				return rows, errors.Wrap(err, "parse slow log failed `"+variable.SlowLogTimeStr+"` error")
```

File: infoschema/slow_log.go [hunk 3]
```diff
@@ -120,14 +120,14 @@ func ParseSlowLog(tz *time.Location, reader *bufio.Reader) ([][]types.Datum, err
 					}
 					err = st.setFieldValue(tz, field, fieldValues[i+1])
 					if err != nil {
-						return rows, err
+						return rows, errors.Wrap(err, "parse slow log filed `"+field+"` error")
```
Comment (zz-jason) [inline infoschema/slow_log.go]: ditto

File: infoschema/slow_log.go [hunk 4]
```diff
@@ -120,14 +120,14 @@ func ParseSlowLog(tz *time.Location, reader *bufio.Reader) ([][]types.Datum, err
...
 					}
 				}
 			} else if strings.HasSuffix(line, variable.SlowLogSQLSuffixStr) {
 				// Get the sql string, and mark the start flag to false.
 				err = st.setFieldValue(tz, variable.SlowLogQuerySQLStr, string(hack.Slice(line)))
 				if err != nil {
-					return rows, err
+					return rows, errors.Wrap(err, "parse slow log filed `"+variable.SlowLogQuerySQLStr+"` error")
```
Comment (zz-jason) [inline infoschema/slow_log.go]: ditto

File: infoschema/slow_log.go [hunk 5]
```diff
@@ -274,54 +248,31 @@ func (st *slowQueryTuple) setFieldValue(tz *time.Location, field, value string)
...
-		}
-		st.memMax = num
+		st.memMax, err = strconv.ParseInt(value, 10, 64)
+	case variable.SlowLogSucc:
+		st.succ, err = strconv.ParseBool(value)
 	case variable.SlowLogQuerySQLStr:
 		st.sql = value
 	}
+	if err != nil {
+		return errors.Wrap(err, "parse slow log filed `"+field+"` error")
```
Comment (zz-jason) [inline infoschema/slow_log.go]:
```suggestion
		return errors.Wrap(err, "parse slow log filed `"+faield+"` error")
```

File: infoschema/slow_log.go (hunk 1 above)
Comment (crazycs520) [reply infoschema/slow_log.go]: done.

**Comment by zz-jason** [comment]
@crazycs520 please follow https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md to refine the PR title.

File: infoschema/slow_log_test.go [hunk 6]
```diff
@@ -159,3 +172,17 @@ select * from t;`)
 	_, err = infoschema.ParseSlowLog(loc, scanner)
 	c.Assert(err, IsNil)
 }
+
+func (s *testSuite) TestSlowLogLongQuery(c *C) {
```
Comment (crazycs520) [inline infoschema/slow_log_test.go]: Sorry, This test is redundant, I will remove this test.

**Pull request merged**